src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -14,23 +14,62 @@ public partial class MemoryMappedFile : IDisposable
         private readonly FileStream? _fileStream;
         internal const int DefaultSize = 0;
 
-        // Private constructors to be used by the factory methods.
-        private MemoryMappedFile(SafeMemoryMappedFileHandle handle)
+        /// <summary>
+        /// Initializes a new instance of a memory-mapped file using an existing safe handle.
+        /// </summary>
+        /// <param name="handle">A safe handle that represents a memory-mapped file for sequential access.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="handle" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentException"><paramref name="handle" /> is closed or invalid.</exception>
+        public MemoryMappedFile(SafeMemoryMappedFileHandle handle)
         {
-            Debug.Assert(handle != null);
-            Debug.Assert(!handle.IsClosed);
-            Debug.Assert(!handle.IsInvalid);
+            if (handle == null)
+            {
+                throw new ArgumentNullException(nameof(handle));
+            }
+
+            if (handle.IsClosed)
+            {
+                throw new ArgumentException(SR.Argument_HandleIsClosed);
+            }
+
+            if (handle.IsInvalid)
+            {
+                throw new ArgumentException(SR.Argument_HandleIsInvalid);
+            }
 
             _handle = handle;
             _leaveOpen = true; // No FileStream to dispose of in this case.
         }
 
-        private MemoryMappedFile(SafeMemoryMappedFileHandle handle, FileStream fileStream, bool leaveOpen)
+        /// <summary>
+        /// Initializes a new instance of a memory-mapped file using an existing safe handle, the file stream of the , and the option to leave the file stream open.
+        /// </summary>
+        /// <param name="handle">A safe handle that represents a memory-mapped file for sequential access.</param>
+        /// <param name="fileStream">A stream opened for the file.</param>
+        /// <param name="leaveOpen">Whether the file stream should be left open or not.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="handle" /> or <paramref name="fileStream" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentException"><paramref name="handle" /> is closed or invalid.</exception>
+        public MemoryMappedFile(SafeMemoryMappedFileHandle handle, FileStream fileStream, bool leaveOpen)
```

Comment:
Update: Before I publish this draft PR, I will have to bring this other private constructor to API Approval to make it public. 

This constructor, along with the other one that receives only the handle (and has already been approved to be made public), are needed internally when creating MemoryMappedFiles via CreateOrOpen, CreateNew, CreateFromFile.